### ISSUE TITLE
CloudLinux guest tools installation support

### ISSUE THREAD

**Issue by iServU** [xcp-ng/xcp#77]
I run a client website hosting VM using CloudLinux to allow for client usage restrictions at the kernel level.

Current OS version is:

When I try to install the guest tools using CentOS which is what CloudLinux (i think) is based on, I receive errors saying that the host OS is not recognized.

Here is the process I followed:
[root@cpsvr01 glopez]# mount /dev/cdrom /mnt
mount: /dev/sr0 is write-protected, mounting read-only

Then:

[root@cpsvr01 glopez]# /mnt/Linux/install.sh
Fatal Error: Failed to determine Linux distribution and version.

**Comment by iServU** [comment]
Any chance you can just send me a link of where to post it and I’ll gladly do it again? I’m extremely tight for time as of late and not very familiar with GitHub.

Many thanks.



Sincerely,


Gabriel Lopez
Chief Technologist
- - - - - - - - - - - - - - - - - - - - - - - - - - - -
Galonet Tech Corp
Miami, FL, United States
www.galonet.com<http://www.galonet.com/>
[http://www.galonet.com/galonet.com.logo.email.png]<http://www.galonet.com/galonet.com.logo.email.png>

**Comment by stormi** [comment]
I don't understand your answer. I just gave you a tip to make it work right now :)

**Comment by iServU** [comment]
LOL, I tried running it with CentOS manual RPM install options but it failed. Any chance you could provide me with the commands as the Google search I found wasn’t very thorough 😊






Sincerely,


Gabriel Lopez
Chief Technologist
- - - - - - - - - - - - - - - - - - - - - - - - - - - -
Galonet Tech Corp
Miami, FL, United States
www.galonet.com<http://www.galonet.com/>
[http://www.galonet.com/galonet.com.logo.email.png]<http://www.galonet.com/galonet.com.logo.email.png>

**Comment by stormi** [comment]
I have never used CloudLinux so I don't know if the RPM for CentOS is compatible with it.

You could try `/mnt/Linux/install.sh -d centos -m 7` to make the installer script behave as if you were in CentOS 7, but really can't guarantee anything about it because I don't know CloudLinux.

If it doesn't work, you can try to install the contents from xe-guest-utilities_7.10.0-1_all.tgz manually.

Else, we'll try to look at it but I can't say when exactly.

**Comment by stormi** [comment]
You could also try the forum, maybe someone from our community made it work in the past already : https://xcp-ng.org/forum/

**Comment by iServU** [comment]
Great news, this workaround worked perfectly! However, for obvious reasons, Operating System isn’t detected in XCP-ng Center 😊

Many thanks Samuel!



Sincerely,


Gabriel Lopez
Chief Technologist
- - - - - - - - - - - - - - - - - - - - - - - - - - - -
Galonet Tech Corp
Miami, FL, United States
www.galonet.com<http://www.galonet.com/>
[http://www.galonet.com/galonet.com.logo.email.png]<http://www.galonet.com/galonet.com.logo.email.png>

**Comment by iServU** [comment]
> 
> 
> I have never used CloudLinux so I don't know if the RPM for CentOS is compatible with it.
> 
> You could try `/mnt/Linux/install.sh -d centos -m 7` to make the installer script behave as if you were in CentOS 7, but really can't guarantee anything about it because I don't know CloudLinux.
> 
> If it doesn't work, you can try to install the contents from xe-guest-utilities_7.10.0-1_all.tgz manually.
> 
> Else, we'll try to look at it but I can't say when exactly.

While this might have worked for me previously, I believe the route was CentOS 6 upgraded to CentOS 7, then to CloudLinux 7.

I just created a new VM and can't get the "Management Agent" to install regardless of the CentOS manual directives used. Tried both install.sh path and RPM package installs.

**Comment by stormi** [comment]
What's the exact output of `/mnt/Linux/install.sh -d centos -m 7`?

**Comment by iServU** [comment]
I get the following:

[root@cpsvr01.iservu.us]:[/mnt/Linux]# /mnt/Linux/install.sh -d centos -m 7
Distribution `centos' version `7' given on command line.

The following changes will be made to this Virtual Machine:
  * packages to be installed/upgraded:
    - xe-guest-utilities-7.10.0-1.x86_64.rpm
    - xe-guest-utilities-xenstore-7.10.0-1.x86_64.rpm

Continue? [y/n] Y

Preparing...                          ################################# [100%]
        package xe-guest-utilities-xenstore-7.10.0-1.x86_64 is already installed
        package xe-guest-utilities-7.10.0-1.x86_64 is already installed

I’ve already tried manual packages install but it still isn’t recognized in XCP-ng Manager properly




Sincerely,


Gabriel Lopez
Chief Technologist
- - - - - - - - - - - - - - - - - - - - - - - - - - - -
Galonet Tech Corp
Miami, FL, United States
www.galonet.com<http://www.galonet.com/>
[cid:image001.png@01D544A2.CFED3E00]<http://www.galonet.com/galonet.com.logo.email.png>

**Comment by borzel** [comment]
@iServU

> "Management Agent"

That therm is only displayed if you used a non linux/unix template for VM creation.

You can create a new VM using the centos template, delete the new disk and attach the disk from your cloudlinux vm.

**Comment by frogstarr78** [comment]
To fix this being displayed correctly in Xen edit this file:

/usr/sbin/xe-linux-distribution

Under this line (approx line number 150):

-e 's/^CentOS Linux release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=centos;major=\1;minor=\2/gp;' \

Add this line:

-e 's/^CloudLinux release \([0-9]*\)\.\([0-9]*\)\(\.[0-9]*\)\? (.*)/distro=cloudlinux;major=\1;minor=\2/gp;' \

**Comment by olivierlambert** [comment]
I suppose we could try a PR directly on top of https://github.com/xenserver/xe-guest-utilities/blob/master/mk/xe-linux-distribution

What do you think @frogstarr78 do you want to try?

**Comment by frogstarr78** [comment]
> I suppose we could try a PR directly on top of https://github.com/xenserver/xe-guest-utilities/blob/master/mk/xe-linux-distribution
> 
> What do you think @frogstarr78 do you want to try?

Absolutely! I've been trying to figure out where to make this PR. Thanks!

**Comment by olivierlambert** [comment]
In Github, it should be entirely "GUI driven": click on "Edit button" (the pencil), this will automatically fork the project in your account, then you'll make the modification, then create a PR with one button and that's it!

**Comment by frogstarr78** [comment]
> In Github, it should be entirely "GUI driven": click on "Edit button" (the pencil), this will automatically fork the project in your account, then you'll make the modification, then create a PR with one button and that's it!
> 
> edit: do not forget to explain the "why" in the commit/PR text field, so Citrix devs will know why you are asking for that.

Nice. Yeah I was thinking I had to pull the whole thing down locally. Also couldn't track down which project the file was in. Thanks again. PR created.

**Comment by iServU** [comment]
Sure, I’ll give it a try and report back.



Sincerely,


Gabriel Lopez
Chief Technologist
- - - - - - - - - - - - - - - - - - - - - - - - - - - -
Galonet Tech Corp
Miami, FL, United States
www.galonet.com<http://www.galonet.com/>
[http://www.galonet.com/galonet.com.logo.email.png]<http://www.galonet.com/galonet.com.logo.email.png>

**Comment by stormi** [comment]
Update published: https://xcp-ng.org/blog/2019/11/08/xcp-ng-security-bulletin-november-2019/
